dbms/src/Flash/tests/gtest_compute_server.cpp

```diff
@@ -885,56 +889,126 @@ try
-        for (const auto & query_id : query_ids)
-            MockComputeServerManager::instance().cancelQuery(query_id);
+        for (const auto & gather_id : gather_ids)
+            MockComputeServerManager::instance().cancelGather(gather_id);
         for (auto & t : running_queries)
             t.join();
         running_queries.clear();
-        query_ids.clear();
+        gather_ids.clear();
         /// case 2, non-min tso can't be added
         for (size_t i = 0; i < active_set_soft_limit; ++i)
         {
-            addOneQuery((i + 1) * 20, running_queries, query_ids);
+            addOneQuery((i + 1) * 20, running_queries, gather_ids);
         }
         using namespace std::literals::chrono_literals;
         std::this_thread::sleep_for(2s);
         ASSERT_TRUE(TiFlashMetrics::instance().tiflash_task_scheduler.get(tiflash_task_scheduler_metrics::type_active_queries_count).Value() == 2);
         ASSERT_TRUE(TiFlashMetrics::instance().tiflash_task_scheduler.get(tiflash_task_scheduler_metrics::type_waiting_queries_count).Value() == 0);
-        addOneQuery(30, running_queries, query_ids);
+        addOneQuery(30, running_queries, gather_ids);
         std::this_thread::sleep_for(2s);
         ASSERT_TRUE(TiFlashMetrics::instance().tiflash_task_scheduler.get(tiflash_task_scheduler_metrics::type_active_queries_count).Value() == 2);
         ASSERT_TRUE(TiFlashMetrics::instance().tiflash_task_scheduler.get(tiflash_task_scheduler_metrics::type_waiting_queries_count).Value() == 1);
         /// cancel 1 running query
-        MockComputeServerManager::instance().cancelQuery(query_ids[0]);
+        MockComputeServerManager::instance().cancelGather(gather_ids[0]);
         running_queries[0].join();
         std::this_thread::sleep_for(2s);
         ASSERT_TRUE(TiFlashMetrics::instance().tiflash_task_scheduler.get(tiflash_task_scheduler_metrics::type_active_queries_count).Value() == 2);
         ASSERT_TRUE(TiFlashMetrics::instance().tiflash_task_scheduler.get(tiflash_task_scheduler_metrics::type_waiting_queries_count).Value() == 0);
         for (size_t i = 1; i < running_queries.size(); i++)
-            MockComputeServerManager::instance().cancelQuery(query_ids[i]);
+            MockComputeServerManager::instance().cancelGather(gather_ids[i]);
         for (size_t i = 1; i < running_queries.size(); i++)
             running_queries[i].join();
     }
     catch (...)
     {
-        for (const auto & query_id : query_ids)
-            MockComputeServerManager::instance().cancelQuery(query_id);
+        for (const auto & gather_id : gather_ids)
+            MockComputeServerManager::instance().cancelGather(gather_id);
+        for (auto & t : running_queries)
+            if (t.joinable())
+                t.join();
+        throw;
+    }
+}
+CATCH
+
+TEST_F(ComputeServerRunner, testCancelMPPGather)
```

Comment:
Can you add an `apply with limit` case to the fullstack test?

Reply:
OK, add some apply tests.